### ISSUE TITLE
Does this work on Mac OS?

### ISSUE THREAD

**Issue by jaybkim1** [DoubleLabyrinth/navicat-keygen#119]
Does this work on Mac OS?

If there is anyone figured out how to get this working on Mac OS, it will be grateful to guide me a little bit.

**Comment by jaybkim1** [comment]
I just found [this opened issue #87](https://github.com/DoubleLabyrinth/navicat-keygen/issues/87).

I will try and update this issue.

**Comment by jaybkim1** [comment]
I am stuck at number 2 (Run the patcher) from [this instruction](https://www.notion.so/The-tutorial-for-cracking-Navicat-Premium-12-on-macOS-1d0a893358e64e8bb3b8b2a7dba993b9)

Anyone knows where the navicat executable file path? 

`$./navicat-patcher <your navicat executable file path>`

**Comment by DoubleLabyrinth** [comment]
Generally speaking, it should be `/Applications/Navicat\ Premium.app/Contents/MacOS/Navicat\ Premium`

You can see an example in `README.md` on `mac` branch

**Comment by lishniy** [comment]
just-pc:navicat-patcher lishniy$ ./navicat-patcher /Applications/Navicat\ Premium.app/
Failed to open file. CODE: 0x00000015
just-pc:navicat-patcher lishniy$ sudo ./navicat-patcher /Applications/Navicat\ Premium.app/
Password:
Failed to open file. CODE: 0x00000015

**Comment by DoubleLabyrinth** [comment]
`/Applications/Navicat\ Premium.app` is actually a folder.

You must specified `/Applications/Navicat\ Premium.app/Contents/MacOS/Navicat\ Premium` which is a path to a Mach-O file, in other word, an executable file like `exe` in Windows.

**Comment by jaybkim1** [comment]
@DoubleLabyrinth Thank you so much for your quick response. 

I successfully followed the instructions in `README.md `on `mac` branch.

For those who are having a trouble following the README.md on mac branch, here is how to use this awesome navicat-keygen in detail.

**- Step 1 - [ Build Keygen ]**  

$ git clone https://github.com/DoubleLabyrinth/navicat-keygen.git -b mac
$ cd navicat-keygen && make release
$ cd navicat-patcher && make release
 
**- Step 2 - [Generate RSA-2048 private key and public key.** 

Since I downloaded Navicat Premium version of 12.1, I decided to use prepared files.

**- Step 3 - [ For Navicat Premium version >= 12.0.24:]** 

I backed up my Navicat Premium.app, but `I skipped this next parts of saving database connection configurations and deleting all of passwords in Keychain.app that is saved by Navicat since I don't have any saved connection configs and passwords in keychain.app that is saved by Navicat.`

>Backup your Navicat Premium.app/Contents/MacOS/Navicat Premium and all of your saved database connection configurations (with password).

>Delete all of passwords in Keychain.app that is saved by Navicat.

**- Step 4 - [ Run patcher ]** 

$ cd navicat-patcher
$ ./navicat-patcher  /Applications/Navicat\ Premium.app/Contents/MacOS/Navicat\ Premium

Result from the command line

> OS/Navicat\ Premium
Open file successfully.
Get file size successfully: 75372000
Map file successfully.
offset = 0x0000000002cb0f57
Success!

**- Step 5 - [ Generate a self-signed code-sign certificate and always trust it. Then use codesign to re-sign Navicat Premium.app. ]** 

I referenced this link to generate my self-signed code-sign certificate.

[How do I generate a self-signed code-sign certificate?](https://stackoverflow.com/questions/27474751/how-can-i-codesign-an-app-without-being-in-the-mac-developer-program/27474942)

Spotlight Search -> 
Choose Keychain Access -> 
Certificate Assistant -> 
Create Certificate ->
Input Name, Identity Type of Self Signed Root, and Certificate Type of Code Signing.

I successfully generate self-signed root certificate.

$ codesign -f -s "My Self-Signed Certificate Name" /Applications/Navicat\ Premium.app/

Result from the command line

> replacing existing signature and "allow" pop up window

**- Step 6 - [ Then goto navicat-keygen folder and in Terminal:]** 

$ cd navicat-keygen
$ ./navicat-keygen 2048key.pem

Result from the command line

> I got a `snKey` and was asked to input my name and my organization.

I waited now from here.

**- Step 7 - [ Open Navicat Premium, find and click Registration. Then input Registration Key by snKey that keygen gave. Then click Activate.:]** 

I input the generated SnKey, and it failed and Navicat asked me do Manual Activation, and I chose it.

**- Step 8 - [ Copy your request code and paste it in keygen. Leave empty line to tell keygen that your input ends (in other words, type Enter at least twice).:]** 

Got back to terminal and input request code that I got from Manual Activiation.

Input Request Code (in Base64), empty line to return: `Request Code`

Result from the command line

> Activiation Code

**- Step 9 - [ Copy your request code and paste it in keygen. Leave empty line to tell keygen that your input ends (in other words, type Enter at least twice).:]** 

It asked "Import Settings To New Navicat Premium? App version 12.1.6" and clicked import.

![screen shot 2018-08-25 at 8 38 45 pm](https://user-images.githubusercontent.com/20435620/44617913-e5e7a080-a8a6-11e8-9646-2ff84274125b.png)

I clicked "allow".
